### PR TITLE
Add lib.inet.smtp to send email messages from storm

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
 import synapse.lib.stormlib.auth as s_stormlib_auth # NOQA
 import synapse.lib.stormlib.cell as s_stormlib_cell # NOQA
 import synapse.lib.stormlib.json as s_stormlib_json  # NOQA
+import synapse.lib.stormlib.smtp as s_stormlib_stix  # NOQA
```
Comment (rakuy0) [inline synapse/cortex.py:52]: Did you mean to have this as `s_stormlib_smtp` instead?

File: synapse/lib/stormlib/smtp.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+@s_stormtypes.registry.registerLib
+class SmtpLib(s_stormtypes.Lib):
+    '''
+    A Storm Library for sending email messages via SMTP.
+    '''
+    _storm_locals = (
+        {'name': 'message', 'desc': 'Construct a new email message.',
+         'type': {'type': 'function', '_funcname': 'message',
+                          'returns': {'type': 'storm:smtp:message',
+                                      'desc': 'The name of the newly created backup.'}}},
```
Comment (mikemoritz) [inline synapse/lib/stormlib/smtp.py:21]: copy/paste error?

File: synapse/lib/stormlib/smtp.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        self.sender = None
+        self.recipients = []
+        self.headers = {}
+        self.bodytext = None
+        self.bodyhtml = None
+        self.attachments = []
+
+        self.locls.update({
+            'send': self.send,
+            'attach': self.send,
```
Comment (mikemoritz) [inline synapse/lib/stormlib/smtp.py:98]: Should be a ref to self.attachments or a method?

File: synapse/cortex.py (hunk 1 above)
Comment (invisig0th) [reply synapse/cortex.py:52]: yep!  thanks!

File: synapse/lib/stormlib/smtp.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+            await asyncio.wait_for(futu, timeout=timeout)
+
+        except asyncio.CancelledError: # pragma: no cover
+            raise
+
+        except Exception as e:
+            return (False, s_common.excinfo(e))
+
+        return (True, None)
```
Comment (mikemoritz) [inline synapse/lib/stormlib/smtp.py:183]: aiosmtplib.send does return a dict, might be useful to return this to the user

File: synapse/lib/stormtypes.py [hunk 5]
```diff
@@ -280,6 +281,12 @@ async def deref(self, name):
     async def _derefGet(self, name):
         return s_common.novalu
 
+    async def _derefGet(self, name):
+        gtor = self.gtors.get(name)
```
Comment (rakuy0) [inline synapse/lib/stormtypes.py:285]: Should we remove the old `_derefGet) that sits right above this one?

File: synapse/lib/stormlib/smtp.py (hunk 4 above)
Comment (invisig0th) [reply synapse/lib/stormlib/smtp.py:183]: Looks like it includes heavy objects that might need translation, but i'll change the return convention for our send() API to returning `(True, {})` on success so we can be flexible in the future.

**Review by mikemoritz — APPROVED** [review]

File: synapse/lib/stormlib/smtp.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        self.bodyhtml = await s_stormtypes.tostr(html)
+
+    async def _getEmailText(self):
+        return self.bodytext
+
+    async def _getEmailHtml(self):
+        return self.bodyhtml
+
+    #TODO
+    #async def attach(self, sha256, name, mime):
```
Comment (jnwatson) [inline synapse/lib/stormlib/smtp.py:130]: Please remove or at least fix formatting (space between pound and beginning)

**Pull request merged**